lib/modules/manager/pep621/processors/pdm.ts

```diff
@@ -64,9 +65,12 @@ export class PdmProcessor implements PyProjectProcessor {
         logger.debug('No pdm.lock found');
         return null;
       }
+
+      const def = parsePyProject(packageFileName, newPackageFileContent);
       const pythonConstraint: ToolConstraint = {
         toolName: 'python',
-        constraint: config.constraints?.python,
+        constraint:
+          def?.project?.['requires-python'] ?? config.constraints?.python,
```

Comment:
Normally configured constraints take precedence unless the package itself is part of the branch updates